src/cudamatrix/cu-kernels.cu

```diff
@@ -1114,6 +1114,35 @@ static void _cuda_matrix_add_elements(Real *data, MatrixDim dim, Real alpha, Mat
 
 template<typename Real>
 __global__
+static void _cuda_matrix_add_indexed_values(Real *data, MatrixDim dim, Real alpha, const Int32Pair* indices, Real* x, int s) { 
+  int i = blockIdx.x * blockDim.x + threadIdx.x;
```

Comment:
You are not changing 'x' so x should be const.
It's preferred even in CUDA code that non-const arguments should be at the end, so unless there is a good reason, 'data' should be the last argument.
Some of the old code may not have been 100% reviewed and may not be fully compliant with these rules.  You can fix the old code too while you're about it, if you have time.